oci/archive/oci_transport.go

```diff
@@ -0,0 +1,117 @@
+package archive
+
+import (
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/image/image"
+	"github.com/containers/image/oci/layout"
+	"github.com/containers/image/transports"
+	"github.com/containers/image/types"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	transports.Register(Transport)
+}
+
+// Transport is an ImageTransport for OCI archive
+// it creates an oci-archive tar file by calling into the OCI transport
+// tarring the directory created by oci and deleting the directory
+var Transport = ociArchiveTransport{}
+
+type ociArchiveTransport struct{}
+
+// ociArchiveReference is an ImageReference for OCI Archive paths
+type ociArchiveReference struct {
+	dir         string
+	resolvedDir string
+	tag         string
+	ociRef      types.ImageReference
+}
+
+func (t ociArchiveTransport) Name() string {
+	return "oci-archive"
+}
+
+// ParseReference converts a string, which should not start with the ImageTransport.Name prefix
+// into an ImageReference.
+func (t ociArchiveTransport) ParseReference(reference string) (types.ImageReference, error) {
+	// ociArchRef stores the dir, resolvedDir, and tag of the original path given by the user
+	ociArchRef, err := layout.Transport.ParseReference(reference)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error parsing reference %q", reference)
+	}
+	dir, resolvedDir, tag := layout.GetOciReference(ociArchRef)
+
+	// tempRef is the path where the directory of the oci image created by oci is stored
+	// this directory is deleted after a tar file is created from it
+	tempRef := "/var/tmp/oci-image:" + tag
+	// ociRef stores the dir, resolvedDir, and tag of the temporary path
+	ociRef, err := layout.Transport.ParseReference(tempRef)
```

Comment:
There is `layout.NewReference` with separate `dir` and `tag` arguments; it isn’t necessary to format them into a string only for `layout.Transport.ParseReference` to split them again.